src/libstd/collections/hash/set.rs

```diff
@@ -618,6 +618,58 @@ impl<T, S> HashSet<T, S>
         self.map.get_key_value(value).map(|(k, _)| k)
     }
 
+    /// Inserts the given `value` into the set if it is not present, then
+    /// returns a reference to the value in the set.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(hash_set_entry)]
+    ///
+    /// use std::collections::HashSet;
+    ///
+    /// let mut set: HashSet<_> = [1, 2, 3].iter().cloned().collect();
+    /// assert_eq!(set.len(), 3);
+    /// assert_eq!(set.get_or_insert(2), &2);
+    /// assert_eq!(set.get_or_insert(100), &100);
+    /// assert_eq!(set.len(), 4); // 100 was inserted
+    /// ```
+    #[inline]
+    #[unstable(feature = "hash_set_entry", issue = "60896")]
+    pub fn get_or_insert(&mut self, value: T) -> &T {
+        self.map.raw_entry_mut().from_key(&value).or_insert(value, ()).0
```

Comment:
Why does this return a shared reference? https://doc.rust-lang.org/nightly/std/collections/hash_map/enum.RawEntryMut.html#method.or_insert gives you two mutable references... am I missing something?

Reply:
I did that to be consistent with plain `get` -- the fact that I used `raw_entry_mut()` is an implementation detail.

We could add `_mut` versions of all these `get` methods if desired, but that is a "raw" thing to do -- you're not supposed to change its `Eq` or `Hash` comparisons. That's also true of interior mutability though.

Reply:
I see; I think this would warrant a comment in the implementation to avoid confusion for other readers in the future.

Reply:
OK, commented -- let me know what you think.

Reply:
Sounds good. 👍